src/main.rs

```diff
@@ -110,60 +114,66 @@ fn handle_error_from(err: rubyfmt::RichFormatError, source: &str, error_exit: Er
+            let mut buffer = String::new();
+            io::stdin()
+                .read_to_string(&mut buffer)
+                .expect("reading from stdin to not fail");
+            let res = rubyfmt::format_buffer(&buffer);
+            match res {
+                Ok(res) => {
+                    write!(io::stdout(), "{}", res).expect("write works");
+                    io::stdout().flush().expect("flush works");
+                }
+                Err(e) => handle_error_from(e, Path::new("stdin"), ErrorExit::Exit),
             }
-            Err(e) => handle_error_from(e, "stdin", ErrorExit::Exit),
         }
-    } else if args.len() == 2 {
-        // consume a filename
-        if let Ok(md) = metadata(args[1].clone()) {
-            if md.is_dir() {
-                format_parts(&[args[1].clone()])
-            } else {
-                let buffer = read_to_string(args[1].clone()).expect("file exists");
-                let res = rubyfmt::format_buffer(&buffer);
-                match res {
-                    Ok(res) => {
-                        write!(io::stdout(), "{}", res).expect("write works");
-                        io::stdout().flush().expect("flush works");
+        // In Rust 1.53
+        // (Some("--help" | "-h"), _) => {
+        (Some("--help"), _) | (Some("-h"), _) => {
+            eprintln!("{}", include_str!("../README.md"));
+            exit(1);
+        }
+        (Some("--internal-fetch-latest-version"), _) => {
+            updates::fetch_latest_version().unwrap();
+        }
+        // Single file
+        (_, [filename]) => {
+            if let Ok(md) = metadata(&filename) {
+                if md.is_dir() {
+                    format_parts(&[filename.clone()])
+                } else {
+                    let buffer = read_to_string(&filename).expect("file exists");
+                    let res = rubyfmt::format_buffer(&buffer);
+                    match res {
+                        Ok(res) => {
+                            write!(io::stdout(), "{}", res).expect("write works");
+                            io::stdout().flush().expect("flush works");
+                        }
+                        Err(e) => handle_error_from(e, filename.as_ref(), ErrorExit::Exit),
                     }
-                    Err(e) => handle_error_from(e, &args[1], ErrorExit::Exit),
                 }
+            } else {
+                eprintln!("{} does not exist", Path::new(&filename).display());
+                exit(1)
             }
-        } else {
-            eprintln!("{} does not exist", args[1]);
-            exit(1)
         }
-    } else if args[1] == "-i" {
-        // inline a file or directory
-        let parts = &args[2..args.len()];
-        format_parts(parts);
-    } else {
-        // inline many files and directories
-        let parts = &args[1..args.len()];
-        format_parts(parts);
+        // Multiple files
+        (Some("-i"), [_, parts @ ..]) | (_, parts) => {
```

Comment:
What does the `@ ..` operator do?

Reply:
`ident @ pat` binds whatever matches `pat` to `ident`. `..` in a pattern is "the rest of it", which in this case is the rest of the slice.